### PR TITLE
[NPU] [Features] [Bugfix] Support mindiesd adaln

### PR DESCRIPTION
**Pull request by jiangmengyu18**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose

1. accelerate adaln by mindiesd.
2. fix bug about  adaln implemented by torch_npu. 

The following is a detailed explanation of the bug about adaln implemented by torch_npu. 

First of all, `torch_npu.npu_layer_norm_eval` is an optimized implementation of `torch.nn.functional.layer_norm` on NPU. The reason for using torch_npu.npu_layer_norm_eval here is that it is exactly equivalent to AdaLN when batch_size = 1, but it may cause precision issues when batch_size > 1.

Since AdaLN normalizes the last dimension of `x`, which is `d`, `normalized_shape` must be `[d]`, and the shapes of `weight` and `bias` must also be `[d]`. However, the shapes of `scale_result` and `shift_result` are `[b, 1, d]`. Although `torch_npu.npu_layer_norm_eval` does not raise an error when the shapes of `weight` and `bias` are `[b, 1, d]`, it will take `weight[0][0]` and `bias[0][0]` for broadcasting, instead of using the full weight and bias. It will lead to precision issues when batch_size > 1.

For example:
init

error result:
error_reuslt

reproduce the error result a second time:
error_reuslt_1

correct reuslt:
correct_result

## Test Plan

- [x] Qwen-Image-Edit-2509

## Test Result
The table below shows the time consumption of AdaLN when using native, torch_npu, and mindiesd, respectively.
| native   | torch_npu   | mindiesd   |
|:---:|:---:|:---:|
| 36.8s | 36.7s | 35.8s |

Performance:
native:
output_native_adaln

torch_npu:
output_torchnpu_adaln

mindiesd:
output_mindiesd_adaln

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan. Please providing the test scripts & test commands. Please state the reasons if your codes don't require additional test scripts. For test file guidelines, please check the [test style doc](https://docs.vllm.ai/projects/vllm-omni/en/latest/contributing/ci/tests_style/)
- [ ] The test results. Please pasting the results comparison before and after, or e2e results.
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model. **Please run `mkdocs serve` to sync the documentation editions to `./docs`.**
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm_omni/diffusion/layers/adalayernorm.py [hunk 1]
```diff
@@ -83,11 +87,27 @@ def forward_npu(
...
         shift_result, scale_result, gate_result = self.preprocess(mod_params, index)
 
-        import torch_npu
+        if _HAS_MINDIESD:
+            try:
+                from mindiesd import layernorm_scale_shift
+
+                output = layernorm_scale_shift(self.layernorm, x, scale_result, shift_result, fused=True)
+            except Exception as e:
+                logger.warning(f"mindiesd import failed, falling back to torch_npu: {e}")
```
Comment (gcanlin) [inline vllm_omni/diffusion/layers/adalayernorm.py]: Can use `logger.warning_once` to fix the Codex comment.

```suggestion
                logger.warning_once(f"mindiesd import failed, falling back to torch_npu: {e}")
```

File: vllm_omni/diffusion/layers/adalayernorm.py [hunk 2]
```diff
@@ -83,11 +87,27 @@ def forward_npu(
...
+        else:
+            import torch_npu
 
-        output = torch_npu.npu_layer_norm_eval(
-            x, normalized_shape=[self.hidden_size], weight=(1 + scale_result), bias=shift_result, eps=self.eps
-        )
+            output = (
+                torch_npu.npu_layer_norm_eval(x, normalized_shape=[self.hidden_size], eps=self.eps) * (1 + scale_result)
+                + shift_result
+            )
```
Comment (gcanlin) [inline vllm_omni/diffusion/layers/adalayernorm.py]: How about removing one of duplicate `output = (torch_npu.npu_layer_norm_eval....)`?

```suggestion
        if _HAS_MINDIESD:
            try:
                from mindiesd import layernorm_scale_shift

                output = layernorm_scale_shift(self.layernorm, x, scale_result, shift_result, fused=True)
                return output, gate_result
            except Exception as e:
                logger.warning_once(f"mindiesd import failed, falling back to torch_npu: {e}")
        
        import torch_npu

        output = (
            torch_npu.npu_layer_norm_eval(x, normalized_shape=[self.hidden_size], eps=self.eps) * (1 + scale_result)
            + shift_result
        )
        return output, gate_result
```

**Review by gcanlin — APPROVED** [review]
Look good now. Thanks!
cc @hsliuustc0106

File: vllm_omni/diffusion/layers/adalayernorm.py [hunk 3]
```diff
@@ -83,10 +87,21 @@ def forward_npu(
...
         shift_result, scale_result, gate_result = self.preprocess(mod_params, index)
 
+        if _HAS_MINDIESD:
+            try:
+                from mindiesd import layernorm_scale_shift
+
+                output = layernorm_scale_shift(self.layernorm, x, scale_result, shift_result, fused=True)
+
+                return output, gate_result
+            except Exception as e:
```
Comment (lishunyang12) [inline vllm_omni/diffusion/layers/adalayernorm.py]: `except Exception` will also swallow runtime errors from `layernorm_scale_shift` itself (e.g. shape mismatch, device error) and silently fall back to the torch_npu path. That makes it hard to notice if mindiesd is actually broken at runtime vs just not importable.

```suggestion
            except ImportError as e:
                logger.warning_once(f"mindiesd import failed, falling back to torch_npu: {e}")
```

**Review by lishunyang12 — COMMENTED** [review]
Bugfix looks right -- the old code passing scale/shift as weight/bias to `npu_layer_norm_eval` was definitely wrong for batch > 1. Left one comment on the exception handling.

File: vllm_omni/diffusion/layers/adalayernorm.py [hunk 4]
```diff
@@ -83,10 +87,22 @@ def forward_npu(
...
+
+                return output, gate_result
+            except ImportError as e:
+                logger.warning_once(f"mindiesd import failed, falling back to torch_npu: {e}")
+                logger.warning_once(f"mindiesd import failed, falling back to torch_npu: {e}")
+
         import torch_npu
 
-        output = torch_npu.npu_layer_norm_eval(
-            x, normalized_shape=[self.hidden_size], weight=(1 + scale_result), bias=shift_result, eps=self.eps
```
Comment (lishunyang12) [inline vllm_omni/diffusion/layers/adalayernorm.py:89]: Duplicated line. Remove one.

**Review by lishunyang12 — COMMENTED** [review]
Bugfix looks correct — applying scale/shift outside `npu_layer_norm_eval` avoids the silent broadcasting issue. Left a nit about a duplicated line.

File: vllm_omni/diffusion/layers/adalayernorm.py (hunk 4 above)
Comment (jiangmengyu18) [reply vllm_omni/diffusion/layers/adalayernorm.py:89]: you mean the rebundant logger.warning_once? I have pull request to solve it.

Comment (jiangmengyu18) [reply vllm_omni/diffusion/layers/adalayernorm.py:89]: cc @lishunyang12
